### PR TITLE
[Improvement][SQL] Performance optimization for a large number(10k) of tasks

### PR DESCRIPTION
**Pull request by reele**
## Purpose of the pull request

Some database optimizations about scheduling

When a project contains 2k of processes and 3k of dependent nodes, the data volume of t_ds_task_instance/t_ds_relation_process_instance/t_ds_process_task_relation_log grows very fast, and running for a few days shows performance issues. Because of the lack of indexes, MasterServer scans dependent nodes and sub_process nodes It will cause serious performance problems in the database (eg: t_ds_task_instance has 200k records caused the 8c16g postgresql server CPU occupies 100%).

After optimization, the CPU usage of the database server is reduced to less than 10%.

## Brief change log

Add 3 indexes to the scheduler's database

### CODE REVIEW

**Review by yimaixinchen — COMMENTED** [review]
can you add the same index in prostgresql.sql?

**Comment by reele** [comment]
> can you add the same index in prostgresql.sql?

I modified:
`/dolphinscheduler-dao/src/main/resources/sql/dolphinscheduler_mysql.sql`,
`/dolphinscheduler-dao/src/main/resources/sql/dolphinscheduler_postgresql.sql`,
`/dolphinscheduler-dao/src/main/resources/sql/upgrade/2.1.0_schema/postgresql/dolphinscheduler_ddl.sql`,
`/dolphinscheduler-dao/src/main/resources/sql/upgrade/2.1.0_schema/mysql/dolphinscheduler_ddl.sql`.

**Comment by reele** [comment]
the conflicts resolved.

**Comment by cwr31** [comment]
Could you show us your performance testing results comparison?

File: dolphinscheduler-dao/src/main/resources/sql/dolphinscheduler_postgresql.sql [hunk 1]
```diff
@@ -489,6 +489,8 @@ CREATE TABLE t_ds_process_task_relation_log (
 
 create index process_task_relation_log_idx_project_code_process_definition_code on t_ds_process_task_relation_log (project_code,process_definition_code);
 
+create index idx_process_code_version on t_ds_process_task_relation_log (process_definition_code, process_definition_version);
```
Comment (caishunfeng) [inline dolphinscheduler-dao/src/main/resources/sql/dolphinscheduler_postgresql.sql:494]: miss this ddl in mysql, please take a check.

Comment (reele) [reply dolphinscheduler-dao/src/main/resources/sql/dolphinscheduler_postgresql.sql:494]: The index added.

**Comment by reele** [comment]
> Could you show us your performance testing results comparison?

Well, The performance problem is obvious, I will try to reproduce the problem.

File: dolphinscheduler-dao/src/main/resources/sql/upgrade/2.1.0_schema/mysql/dolphinscheduler_ddl.sql [hunk 2]
```diff
@@ -20,6 +20,10 @@ ALTER TABLE `t_ds_task_instance` MODIFY COLUMN `task_params` longtext COMMENT 'j
 ALTER TABLE `t_ds_process_task_relation` ADD KEY `idx_code` (`project_code`, `process_definition_code`) USING BTREE;
 ALTER TABLE `t_ds_process_task_relation_log` ADD KEY `idx_process_code_version` (`process_definition_code`,`process_definition_version`) USING BTREE;
 
+ALTER TABLE `t_ds_relation_process_instance` ADD INDEX `idx_relation_process_instance_pproc_inst_id_ptask_inst_id` (`parent_process_instance_id`, `parent_task_instance_id`);
```
Comment (caishunfeng) [inline dolphinscheduler-dao/src/main/resources/sql/upgrade/2.1.0_schema/mysql/dolphinscheduler_ddl.sql:25]: miss the ddl in upgrade too.

Comment (reele) [reply dolphinscheduler-dao/src/main/resources/sql/upgrade/2.1.0_schema/mysql/dolphinscheduler_ddl.sql:25]: this index is already exist at line 21...

**Review by caishunfeng — APPROVED** [review]
Look good to me if ci pass

**Pull request closed without merging**